### PR TITLE
Step towards displaying username at nav bar

### PR DESCRIPTION
**Pull request by anubhabsen**

### CODE REVIEW

File: src/auth.js [hunk 1]
```diff
@@ -4,12 +4,14 @@ var config = require('./config');
...
-function sendToken (user_id, res) {
+function sendToken (user_id, res, user) {
+  user = user.toJSON();
+  user.password = undefined;
   jwt.sign({}, config.secret, { subject: user_id.toString(), expiresIn: "2h" }, function (err, token) {
     if (err != null) {
       throw err;
     } else {
-      res.send({token: token});
+      res.send({token: token, userName: user.name});
```
Comment (meghprkh) [inline src/auth.js]: replace this with 

``` js
res.send({token: token, ...user});
```

or probably just

``` js
res.send({token: token, user});
```

File: src/auth.js [hunk 2]
```diff
@@ -4,12 +4,14 @@ var config = require('./config');
 var models = require('./models');
 var middleware = require('./middleware');
 
-function sendToken (user_id, res) {
+function sendToken (user_id, res, user) {
```
Comment (meghprkh) [inline src/auth.js]: You dont need `user_id` if you are getting `user`.  Change this and all calls to

``` js
function sendToken (user, res)
```

And get by user id by `user.id` then inside `sendToken`

File: src/auth.js (hunk 1 above)
Comment (anubhabsen) [inline src/auth.js]: what exactly does this mean? not exactly object notation is it?

File: src/auth.js [hunk 3]
```diff
@@ -4,14 +4,14 @@ var config = require('./config');
...
+function sendToken (user, res) {
   user = user.toJSON();
   user.password = undefined;
-  jwt.sign({}, config.secret, { subject: user_id.toString(), expiresIn: "2h" }, function (err, token) {
+  jwt.sign({}, config.secret, { subject: user.id.toString(), expiresIn: "2h" }, function (err, token) {
     if (err != null) {
       throw err;
     } else {
-      res.send({token: token, userName: user.name});
+      res.send({token: token, user});
```
Comment (meghprkh) [inline src/auth.js]: Okay I changed my mind a little. As we just need to send email and name, would be better if keep it:

``` js
     res.send({token: token, email: user.email, name: user.password});
```

This way you dont have to worry about hiding values like password from the object. 

Thus you can delete the lines

``` js
user = user.toJSON();
user.password = undefined;
```

File: src/auth.js (hunk 1 above)
Comment (meghprkh) [inline src/auth.js]: Okay `...user` demo http://babeljs.io/repl/#?evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&code=var%20x%20%3D%20%7B%0A%20%20a%3A%201%2C%0A%20%20b%3A%202%2C%0A%20%20c%3A%203%0A%7D%0A%0Avar%20y%20%3D%20%7B%0A%20%20a%3A%2010%2C%0A%20%20...x%2C%0A%20%20b%3A%2020%0A%7D%0A%0Aconsole.log(%27X%27%2C%20x)%0Aconsole.log(%27Y%27%2C%20y)

File: src/auth.js (hunk 3 above)
Comment (anubhabsen) [inline src/auth.js]: `name: user.name` you mean?

**Pull request merged**